include/glow/Base/Tensor.h

```diff
@@ -322,7 +322,10 @@ class Tensor final {
     // If the new size is identical to the allocated size then there is no need
     // to re-allocate the buffer.
     if (type_ == T && getData()) {
-      zero();
+#ifdef GLOW_DEBUG_TENSOR_INIT
```

Comment:
This is a great idea, do you think we should always do it in debug mode (like `#ifndef NDEBUG`)?  Your call either way.

Reply:
Good question. I actually started that way, but I think it covers some heap sentinels written by free in debug. No strong opinions either way, this is a bit more flexible but we don't really want to have to think about this.